source/module.h

```diff
@@ -59,4 +60,45 @@ int aws_napi_callback_init(
 int aws_napi_callback_clean_up(struct aws_napi_callback *cb);
 int aws_napi_callback_dispatch(struct aws_napi_callback *cb, void *user_data);
 
+/*
+ * One of these will be allocated each time the module init function is called
+ * Any global state that isn't thread safe or requires clean up should be stored
+ * on this so that it can be tracked and cleaned up
+ */
+struct aws_napi_context {
+    napi_env env;
+    struct aws_allocator *allocator;
+    struct aws_napi_logger_ctx *logger;
+};
+
+#define _AWS_NAPI_ERROR_MSG(call, file, line) "N-API call failed: " #call " @ " file "(" #line ")"
+#define _AWS_NAPI_PASTE(x) x
+
+/*
+ * AWS_NAPI_CALL(env, napi_xxx(args...), { return NULL; }) will ensure that a failed result is logged as an error
+ */
+#define AWS_NAPI_CALL(env, call, on_fail)                                                                              \
```

Comment:
I'm happy to do a follow up where all napi calls are wrapped in these 2 macros, but this PR is already a nightmare.